server/src/immich/controllers/library.controller.ts

```diff
@@ -18,22 +19,29 @@ import { UUIDParamDto } from './dto/uuid-param.dto';
 @ApiTags('Library')
 @Controller('library')
 @Authenticated()
-@AdminRoute()
 @UseValidation()
 export class LibraryController {
   constructor(private service: LibraryService) {}
 
   @Get()
-  getLibraries(@Auth() auth: AuthDto): Promise<ResponseDto[]> {
+  getUserLibraries(@Auth() auth: AuthDto): Promise<ResponseDto[]> {
     return this.service.getAllForUser(auth);
   }
 
+  @Post('/all')
+  @AdminRoute()
+  getAllLibraries(@Auth() auth: AuthDto, @Body() dto: SearchLibraryDto): Promise<ResponseDto[]> {
+    return this.service.getAll(auth, dto);
+  }
+
```

Comment:
We definitely should not use a POST for a GET. Alternatively we could use a query param to differentiate between getting my libraries vs getting all the libraries.

Reply:
I think I've reworked this now